plugins/module_utils/linode_helper.py

```diff
@@ -305,3 +305,33 @@ def poll_condition(
         step=step,
         timeout=timeout,
     )
+
+
+def get_resource_safe(list_func: Callable[[], List[Any]]) -> Any:
+    """
+    Wraps a resource list function with error handling.
+    If no entries are returned, this function returns None rather than
+    raising an error.
+    """
+    try:
+        list_results = list_func()
+        return None if len(list_results) < 1 else list_results[0]
```

Comment:
Would it be fine to let the caller of this function to check whether the list is empty or not? I think empty list won't raise an error, right?